components/layout_2020/fragment_tree/box_fragment.rs

```diff
@@ -192,6 +192,35 @@ impl BoxFragment {
         )
     }
 
+    pub fn clip_overflow(&self, containing_block_rect: &PhysicalRect<Au>) -> PhysicalRect<Au> {
+        let mut clip_rect = self
+            .padding_rect()
+            .translate(containing_block_rect.origin.to_vector());
+
+        // Adjust by the overflow clip margin.
+        // https://drafts.csswg.org/css-overflow/#overflow-clip-margin
+        let clip_margin = Au::from(self.style.get_margin().overflow_clip_margin);
+        clip_rect.size.width += clip_margin;
+        clip_rect.size.height += clip_margin;
+
+        let mut overflow = self
+            .scrollable_overflow()
+            .translate(containing_block_rect.origin.to_vector());
+
+        let overflow_style = self.style.effective_overflow();
+        if overflow_style.x == ComputedOverflow::Clip {
+            overflow.origin.x = overflow.origin.x.max(clip_rect.origin.x);
+            overflow.size.width = overflow.size.width.min(clip_rect.size.width);
```

Comment:
The logic seems wrong, I think it should just intersect, right? There is `insersection()` but we need it per axis.

So it should take the max of the `min_x()` and the min of the `max_x()` offsets?

Reply:
I'm confused. Why are we intersecting the clip rect with the overflow here?